src/lib.rs

```diff
@@ -24,6 +24,7 @@ extern crate libsystemd_sys;
 extern crate libc;
 extern crate slog;
 
+#[allow(deprecated, unused_imports)]
```

Comment:
Why?

Reply:
As of [Rust 1.24 (2018-02-15)](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1240-2018-02-15), `char.to_ascii_uppercase` has existed directly on char. Because rust defaults to the inherent impl, this import is unused on rust versions >= 1.24.

Later, in [Rust 1.26 (2018-05-10)](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1260-2018-05-10), the `std::ascii::AsciiExt` trait was deprecated.

I didn't do anything which would require a bump of the minimum rustc version of this crate, but didn't want the warnings in the newer rustc versions. Alternatively, if bumping the minimum version is OK, we could also replace the `try!` calls.